pkg/admissioncontroller/webhook/admission/updaterestriction/handler.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package updaterestriction
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+// Handler handles resources that should not be modified
+// from users other than the gardenlet.
+type Handler struct{}
+
+// Handle checks if the request is issued by a gardenlet
+// and rejects the request if that is not the case.
+func (h *Handler) Handle(_ context.Context, req admission.Request) admission.Response {
+	if !slices.Contains(req.UserInfo.Groups, v1beta1constants.SeedsGroup) {
```

Comment:
Does it make sense to handle the restriction generically?

Extensions with Garden access might benefit from this protection as well, but operate with their own service account and thus are not part of the `gardener.cloud:system:seeds` group. How about if it would be possible to additionally configure the group with an annotation (default can still be `gardener.cloud:system:seeds` if non is given)?

Reply:
That is an interesting idea, however I would vote to hold it off unless we have a concrete use case. Do you have any in mind? How can we regulate who is able to set this annotation? I mean if a user wants to annotate and restrict a resource they would have to set label and annotation on it, but then if this resource should not be restricted by users how can we recover from that?

Reply:
I thought it might be a quick win but understand that this can quickly end in deadlocks. Agreeing to hold it off then 👍